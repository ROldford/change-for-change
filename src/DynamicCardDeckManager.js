// Generate an appropriate number of decks for the current window size

import React from 'react';
import { chunk } from 'lodash';
import DynamicCardDeck from './DynamicCardDeck';

export default class DynamicCardDeckManager extends React.Component {
  createDeck(cards, deckSize) {
    let key = cards[0].key;
    return <DynamicCardDeck key={key} cards={cards} deckSize={deckSize} />;
  }

  createDecks(cards, deckSize) {
    const chunkedCards = chunk(cards, deckSize);
    return chunkedCards.map((chunk) => {return this.createDeck(chunk, deckSize)})
  }

  render() {
    return (
      <div>
        {this.createDecks(this.props.cards, this.props.deckSize)}
      </div>
    );
  }
}
