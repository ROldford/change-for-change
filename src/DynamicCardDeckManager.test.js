import DynamicCardDeckManager from "./DynamicCardDeckManager";

const setup = () => {
  return [
    {
      key: 1,
      data: {
        imageSource: 'https://www.countryflags.io/ca/flat/64.png',
        imageAltText: 'Card image cap',
        cardTitle: '1 Card title',
        cardText: 'This is a longer card with supporting text below as a natural lead-in to additional content. This content is a little bit longer.'
      }
    },
    {
      key: 2,
      data: {
        imageSource: 'https://www.countryflags.io/aa/flat/64.png',
        imageAltText: 'Card image cap',
        cardTitle: '2 Card title',
        cardText: 'This is a longer card with supporting text below as a natural lead-in to additional content. This content is a little bit longer.'
      }
    },
    {
      key: 3,
      data: {
        imageSource: 'https://www.countryflags.io/ca/flat/64.png',
        imageAltText: 'Flag: Canada',
        cardTitle: '3 Card title',
        cardText: 'This is a longer card with supporting text below as a natural lead-in to additional content. This content is a little bit longer.'
      }
    },
    {
      key: 4,
      data: {
        imageSource: 'https://www.countryflags.io/ca/flat/64.png',
        imageAltText: 'Flag: Canada',
        cardTitle: '4 Card title',
        cardText: 'This is a longer card with supporting text below as a natural lead-in to additional content. This content is a little bit longer.'
      }
    },
    {
      key: 5,
      data: {
        imageSource: 'https://www.countryflags.io/ca/flat/64.png',
        imageAltText: 'Flag: Canada',
        cardTitle: '5 Card title',
        cardText: 'This is a longer card with supporting text below as a natural lead-in to additional content. This content is a little bit longer.'
      }
    },
    {
      key: 6,
      data: {
        imageSource: 'https://www.countryflags.io/ca/flat/64.png',
        imageAltText: 'Flag: Canada',
        cardTitle: '6 Card title',
        cardText: 'This is a longer card with supporting text below as a natural lead-in to additional content. This content is a little bit longer.'
      }
    },
    {
      key: 7,
      data: {
        imageSource: 'https://www.countryflags.io/ca/flat/64.png',
        imageAltText: 'Flag: Canada',
        cardTitle: '7 Card title',
        cardText: 'This is a longer card with supporting text below as a natural lead-in to additional content. This content is a little bit longer.'
      }
    },
    {
      key: 8,
      data: {
        imageSource: 'https://www.countryflags.io/ca/flat/64.png',
        imageAltText: 'Flag: Canada',
        cardTitle: '8 Card title',
        cardText: 'This is a longer card with supporting text below as a natural lead-in to additional content. This content is a little bit longer.'
      }
    },
    {
      key: 9,
      data: {
        imageSource: 'https://www.countryflags.io/ca/flat/64.png',
        imageAltText: 'Flag: Canada',
        cardTitle: '9 Card title',
        cardText: 'This is a longer card with supporting text below as a natural lead-in to additional content. This content is a little bit longer.'
      }
    },
    {
      key: 10,
      data: {
        imageSource: 'https://www.countryflags.io/ca/flat/64.png',
        imageAltText: 'Flag: Canada',
        cardTitle: '10 Card title',
        cardText: 'This is a longer card with supporting text below as a natural lead-in to additional content. This content is a little bit longer.'
      }
    }
  ];
}

describe('Tests', () => {
  it("renders correctly", () => {
    let cards = setup();
    const wrapper = shallow(
      <DynamicCardDeckManager cards={cards} deckSize={5} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  // it("renders blank cards when needed", () => {
  //   let cards = setup();
  //   const wrapper = shallow(
  //     <DynamicCardDeck cards={cards} deckSize={5} />
  //   )
  //   expect(wrapper).toMatchSnapshot();
  //   const foundCards = wrapper.find('CustomCard');
  //   expect(foundCards.length).toEqual(5);
  //   // const placeholderCards = foundCards.filterWhere((card) => {
  //   //   card.props().data.imageSource === 'https://www.countryflags.io/aa/flat/64.png'
  //   // });
  //   // expect(placeholderCards.length).toEqual(2);
  // });
})
