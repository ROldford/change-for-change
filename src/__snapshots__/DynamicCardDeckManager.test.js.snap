// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests renders correctly 1`] = `
<div>
  <DynamicCardDeck
    cards={
      Array [
        Object {
          "data": Object {
            "cardText": "This is a longer card with supporting text below as a natural lead-in to additional content. This content is a little bit longer.",
            "cardTitle": "1 Card title",
            "imageAltText": "Card image cap",
            "imageSource": "https://www.countryflags.io/ca/flat/64.png",
          },
          "key": 1,
        },
        Object {
          "data": Object {
            "cardText": "This is a longer card with supporting text below as a natural lead-in to additional content. This content is a little bit longer.",
            "cardTitle": "2 Card title",
            "imageAltText": "Card image cap",
            "imageSource": "https://www.countryflags.io/aa/flat/64.png",
          },
          "key": 2,
        },
        Object {
          "data": Object {
            "cardText": "This is a longer card with supporting text below as a natural lead-in to additional content. This content is a little bit longer.",
            "cardTitle": "3 Card title",
            "imageAltText": "Flag: Canada",
            "imageSource": "https://www.countryflags.io/ca/flat/64.png",
          },
          "key": 3,
        },
        Object {
          "data": Object {
            "cardText": "This is a longer card with supporting text below as a natural lead-in to additional content. This content is a little bit longer.",
            "cardTitle": "4 Card title",
            "imageAltText": "Flag: Canada",
            "imageSource": "https://www.countryflags.io/ca/flat/64.png",
          },
          "key": 4,
        },
        Object {
          "data": Object {
            "cardText": "This is a longer card with supporting text below as a natural lead-in to additional content. This content is a little bit longer.",
            "cardTitle": "5 Card title",
            "imageAltText": "Flag: Canada",
            "imageSource": "https://www.countryflags.io/ca/flat/64.png",
          },
          "key": 5,
        },
      ]
    }
    deckSize={5}
    key="1"
  />
  <DynamicCardDeck
    cards={
      Array [
        Object {
          "data": Object {
            "cardText": "This is a longer card with supporting text below as a natural lead-in to additional content. This content is a little bit longer.",
            "cardTitle": "6 Card title",
            "imageAltText": "Flag: Canada",
            "imageSource": "https://www.countryflags.io/ca/flat/64.png",
          },
          "key": 6,
        },
        Object {
          "data": Object {
            "cardText": "This is a longer card with supporting text below as a natural lead-in to additional content. This content is a little bit longer.",
            "cardTitle": "7 Card title",
            "imageAltText": "Flag: Canada",
            "imageSource": "https://www.countryflags.io/ca/flat/64.png",
          },
          "key": 7,
        },
        Object {
          "data": Object {
            "cardText": "This is a longer card with supporting text below as a natural lead-in to additional content. This content is a little bit longer.",
            "cardTitle": "8 Card title",
            "imageAltText": "Flag: Canada",
            "imageSource": "https://www.countryflags.io/ca/flat/64.png",
          },
          "key": 8,
        },
        Object {
          "data": Object {
            "cardText": "This is a longer card with supporting text below as a natural lead-in to additional content. This content is a little bit longer.",
            "cardTitle": "9 Card title",
            "imageAltText": "Flag: Canada",
            "imageSource": "https://www.countryflags.io/ca/flat/64.png",
          },
          "key": 9,
        },
        Object {
          "data": Object {
            "cardText": "This is a longer card with supporting text below as a natural lead-in to additional content. This content is a little bit longer.",
            "cardTitle": "10 Card title",
            "imageAltText": "Flag: Canada",
            "imageSource": "https://www.countryflags.io/ca/flat/64.png",
          },
          "key": 10,
        },
      ]
    }
    deckSize={5}
    key="6"
  />
</div>
`;
